File: ruby/easy/reverse_vowels_of_a_string.rb
# @param {String} s
# @return {String}
def reverse_vowels(s)
    i,j = 0,s.length-1
    while i < j
        if is_vowel(s[i]) && is_vowel(s[j])
            t = s[i]
            s[i] = s[j]
            s[j] = t
            i += 1 and j -= 1
        elsif is_vowel(s[i])
            j -= 1
        elsif is_vowel(s[j])
            i += 1
        else
            i += 1 and j -= 1
        end
    end
    return s
end

def is_vowel(c)
    return "aeiouAEIOU".index(c)
end

File: javascript/easy/reverseVowelsOfAString.js
/**
 * @param {string} s
 * @return {string}
 */
var reverseVowels = function(s) {
    let i = 0, j = s.length-1, res = s.split("");
    while(i < j) {
        if(isVowel(res[i]) && isVowel(res[j])) {
            swap(res, i, j);
            i++; j--;
        } else if(isVowel(res[i])) {
            j--;
        } else if(isVowel(res[j])) {
            i++;
        } else {
            i++; j--;
        }
    }
    return res.join("");
};

var isVowel = function(c) {
    return "aeiouAEIOU".indexOf(c) != -1;
};

var swap = function(arr, i, j) {
    let temp = arr[i];
    arr[i] = arr[j];
    arr[j] = temp;
};
